augly/image/functional.py

```diff
@@ -1786,3 +1787,64 @@ def vflip(
     imutils.get_metadata(metadata=metadata, function_name="vflip", **func_kwargs)
 
     return imutils.ret_and_save_image(aug_image, output_path)
+
+
+def xskew(
+    image: Union[str, Image.Image],
+    output_path: Optional[str] = None,
+    metadata: Optional[List[Dict[str, Any]]] = None,
+) -> Image.Image:
+    """
+    Shearing or Skewing an image with respect to x-axis
+
+    @param image: the path to an image or a variable of type PIL.Image.Image
+        to be augmented
+
+    @param output_path: the path in which the resulting image will be stored.
+        If None, the resulting PIL Image will still be returned
+
+    @param metadata: if set to be a list, metadata about the function execution
+        including its name, the source & dest width, height, etc. will be appended
+        to the inputted list. If set to None, no metadata will be appended or returned
+
+    @returns: the augmented PIL Image
+    """
+    image = imutils.validate_and_load_image(image)
+    rows, cols, dim = image.shape
+    M = np.float32([[1, 0.5, 0],
+                    [0, 1  , 0],
+                    [0, 0  , 1]])             
+    xsheared_img = cv2.warpPerspective(image, M, (int(cols*1.5), int(rows*1.5)))
+
+    return imutils.ret_and_save_image(sheared_img, output_path)
+
+
+def yskew(
+    image: Union[str, Image.Image],
+    output_path: Optional[str] = None,
+    metadata: Optional[List[Dict[str, Any]]] = None,
+) -> Image.Image:
+    """
+    Shearing or Skewing an image with respect to y-axis
+
+    @param image: the path to an image or a variable of type PIL.Image.Image
+        to be augmented
+
+    @param output_path: the path in which the resulting image will be stored.
+        If None, the resulting PIL Image will still be returned
+
+    @param metadata: if set to be a list, metadata about the function execution
+        including its name, the source & dest width, height, etc. will be appended
+        to the inputted list. If set to None, no metadata will be appended or returned
+
+    @returns: the augmented PIL Image
+    """
+    image = imutils.validate_and_load_image(image)
```

Comment:
```suggestion
    func_kwargs = imutils.get_func_kwargs(metadata, locals())
    
    image = imutils.validate_and_load_image(image)
```

Reply:
thank you

Reply:
Thanks for making this change. Actually, can you reverse the order of those 2 lines though? The `image` variable needs to be included in the func kwargs. Thanks!